client/src/components/ApplicationList.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+import { Job, Application } from '../lib';
+import JobApplicationCard from './JobApplicationCard';
+import Pagination from './Pagination';
+
+type ApplicationListProps = {
+  jobApplication: (Application & Job)[] | undefined;
+};
+
+export default function ApplicationList({
+  jobApplication,
```

Comment:
One thing you can do to simplify your code, if you still want to allow `undefined` as a value for `jobApplication`, is to provide a default value:
```
export default function ApplicationList({ jobApplication = [] }: ApplicationListProps) {}
```

Then you won't need to protect against `undefined` in so many places down below.